src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,232 @@
+import {
+  Directive,
+  forwardRef,
+  ElementRef,
+  EventEmitter,
+  Input,
+  HostListener,
+  AfterViewInit,
+  Injector,
+  ChangeDetectorRef,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  AbstractControl,
+  NgControl,
+  FormControl
+} from '@angular/forms';
+
+import {
+  SkyCountryData
+} from './types';
+
+import {
+  SkyPhoneFieldAdapterService
+} from './phone-field-adapter.service';
+
+import {
+  SkyPhoneFieldComponent
+} from './phone-field.component';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_PHONE_FIELD_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+
+const SKY_PHONE_FIELD_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR,
+    SkyPhoneFieldAdapterService
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  /**
+   * Disabling linting here as per the Angular style guide "Style 05-13" it is acceptable
+   * to alias when when the directive name is also an input property,
+   * and the directive name doesn't describe the property.
+   */
+  // tslint:disable-next-line:no-input-rename
+  @Input('skyPhoneFieldInput')
+  public skyPhoneFieldComponent: SkyPhoneFieldComponent;
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public noValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChanged = new EventEmitter<SkyCountryData>();
```

Comment:
Does this need an `ngDestroy` to handle closing? Also, if this is intended to be an emit that fires multiple times, it should be named in the present tense: `selectedCountryChange`.

Reply:
Not for the event emitter. That will destroy automatically. However, I did miss needing to add a `takeUntil` on the directive where it subscribes to that event. I've added that.

I also renamed the event.

Reply:
Actually, I went ahead and completed the event.